src/credentials.cpp

```diff
@@ -214,33 +216,37 @@ namespace preferences
 	}
 }
 
+const int AES_KEY_SIZE = 256;
+
 // TODO: Key-stretching (bcrypt was recommended)
 std::string build_key(const std::string& server, const std::string& login)
 {
 	std::ostringstream out;
 	out << login << get_system_username() << server;
-	return out.str();
-}
-
-// FIXME: XOR encryption is a really terrible choice - swap it out for something better!
-// TODO: Maybe use cryptopp or something for AES encryption?
-static std::string xor_crypt(std::string text, const std::string& key)
-{
-	const size_t m = key.size();
-	for(size_t i = 0; i < text.size(); i++) {
-		text[i] ^= key[i % m];
+	while(out.tellp() < AES_KEY_SIZE / 8) {
+		out.put('x');
 	}
-	return text;
+	return out.str();
 }
 
 std::string encrypt(const std::string& text, const std::string& key)
 {
-	return xor_crypt(text, key);
+	std::shared_ptr<unsigned char> out(new unsigned char[text.size() + 1], std::default_delete<unsigned char[]>());
```

Comment:
I'd prefer if you used a vector for this. You can get the raw pointer by calling `std::vector::data()`.